analytic_engine/src/wal_replicator.rs

```diff
@@ -0,0 +1,380 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! WAL Replicator implementation.
```

Comment:
Rename `Replicator` to `Consumer`.

Reply:
Could you please describe why `Consumer` is better than `Replicator`? My reason for using `Replicator` is stated in #175. TL;DR: A replicator replicates logs from WAL to tables, it keeps the data between `Leader` table and its replicas in sync.

Reply:
`WAL` is divided into `log replicator` and `log consumer`. Currently, the log replicator is implemented by wal mod such as `WAL based on OBKV`, and followers only need to consume logs. In `start_replicate` func, actually calls `consume_log` not `replicate_log`.

Reply:
Or I can interpret `Consumer` as the one that eats logs and generate new data, which is `Table` in `Instance`. Your explanation might be right but I don't think it's better. And if so then we can keep it.

Furthermore in our model, distributed WAL component like `OBKV` is a place to store the write ahead logs. It doesn't "replicate" logs to other nodes neither by definition nor in practice. It's those followers that actually copy logs from storage to itself. IMHO calling this copy procedure "replicate" has no problem.

Reply:
@ShiKaiWi  what you think?

Reply:
Here are my some thoughts about this naming issue:
- `Consumer` is actually more suitable for a more common case (only read logs without knowing how to apply these log contents to table).
- `Replicator` basically denotes makes a redundant copy of the data without changes about the original data, but here the data has been changed into `Table`'s data from the log entries.

So I propose a new name `Synchronizer`: synchronize logs from WAL to table.

Reply:
`Synchronizer` looks good to me. How about @chunshao90